godtools/App/Services/Renderer/Views/Tool/Views/Modal/ToolPageModalView.swift

```diff
@@ -87,18 +87,17 @@ class ToolPageModalView: MobileContentView, NibBased {
         centerContentStackVerticallyIfNeeded()
     }
     
-    override func didReceiveEvents(eventIds: [EventId]) {
-            
-        let eventsContainFollowUp: Bool = eventIds.contains(EventId.Companion().FOLLOWUP)
+    override func didReceiveEvent(eventId: EventId, eventIdsGroup: [EventId]) -> ProcessedEventResult? {
         
-        for eventId in eventIds {
-            
-            if viewModel.listeners.contains(eventId) && !eventsContainFollowUp {
-                delegate?.toolPageModalListenerActivated(modalView: self)
-            }
-            else if viewModel.dismissListeners.contains(eventId) {
-                delegate?.toolPageModalDismissListenerActivated(modalView: self)
-            }
+        let isFollowUpEvent: Bool = eventId == EventId.Companion().FOLLOWUP
```

Comment:
:( I'm sad that this code requires `Companion()`. I'm not sure of a way around that in kotlin native currently

Reply:
Yeah I know, it's a bit clunky